src/main/java/reactor/netty/resources/PooledConnectionProvider.java

```diff
@@ -64,45 +65,26 @@
 import reactor.util.context.Context;
 
 import static reactor.netty.ReactorNetty.format;
+import static reactor.netty.resources.ConnectionProvider.ConnectionPoolSpec.PENDING_ACQUIRE_MAX_COUNT_NOT_SPECIFIED;
 
 /**
  * @author Stephane Maldini
  * @author Violeta Georgieva
  */
 final class PooledConnectionProvider implements ConnectionProvider {
 
-	interface PoolFactory {
-
-		InstrumentedPool<PooledConnection> newPool(Publisher<PooledConnection> allocator);
-	}
-
 	final ConcurrentMap<PoolKey, InstrumentedPool<PooledConnection>> channelPools =
 			PlatformDependent.newConcurrentHashMap();
-	final String      name;
-	final int         maxConnections;
-	final int         pendingAcquireMaxCount;
-	final long        pendingAcquireTimeout;
-	final long        maxIdleTime;
-	final long        maxLifeTime;
-	final boolean     metricsEnabled;
-	final PoolFactory poolFactory;
+	final String                          name;
+	final Map<SocketAddress, PoolFactory> poolFactoryPerRemoteHost = new HashMap<>();
```

Comment:
I'm wondering about the implication of keeping `SocketAddress` references in this map's keyset, but no limited resource should leak from that, I think?